src/App.tsx

```diff
@@ -1,142 +1,298 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import cn from 'classnames';
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { TodoList } from './components/TodoList';
+import { FilterList } from './components/FilterList';
+import {
+  changeCompleteTodo,
+  getTodos,
+  pushTodo,
+  deleteTodo,
+  changeTitleTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Filter } from './types/Filter';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+  const [visibleTodoList, setVisibleList] = useState<Todo[]>([]);
```

Comment:
no need to create a separate state for visible todos, you can count it based on `todoList`. It's better to save the filter value in state and get visibleTodos in this way: `const visibleTodos = filterTodos(todoList)` (less count of re-renders)

In places where you update visible todos (after update 1 todo, for example) change `todoList` instead - after `todoList` will be changed visibleTodos will be recalculated again.